lib/collection/src/recommendations.rs

```diff
@@ -51,36 +53,43 @@ fn avg_vectors<'a>(vectors: impl IntoIterator<Item = VectorRef<'a>>) -> Collecti
                 sparse_count += 1;
                 avg_sparse = vector.combine_aggregate(&avg_sparse, |v1, v2| v1 + v2);
             }
-            VectorRef::MultiDense(_) => {
-                // TODO(colbert)
-                return Err(CollectionError::bad_input(
-                    "MultiDenseVector is not supported".to_owned(),
+            VectorRef::MultiDense(vector) => {
+                multi_count += 1;
+                avg_multi = Some(avg_multi.map_or_else(
+                    || vector.to_owned(),
+                    |mut avg_multi| {
+                        avg_multi
+                            .flattened_vectors
+                            .extend_from_slice(vector.flattened_vectors);
+                        avg_multi
+                    },
                 ));
             }
         }
     }
 
-    match (dense_count, sparse_count) {
+    match (dense_count, sparse_count, multi_count) {
         // TODO(sparse): what if vectors iterator is empty? We added CollectionError::BadRequest,
         // but it's not clear if it's the best solution.
         // Currently it's hard to return an zeroed vector, because we don't know its type: dense or sparse.
-        (0, 0) => Err(CollectionError::bad_input(
+        (0, 0, 0) => Err(CollectionError::bad_input(
             "Positive vectors should not be empty with `average` strategy".to_owned(),
         )),
-        (_, 0) => {
+        (_, 0, 0) => {
             for item in &mut avg_dense {
                 *item /= dense_count as VectorElementType;
             }
-            Ok(avg_dense.into())
+            Ok(Vector::from(avg_dense))
         }
-        (0, _) => {
+        (0, _, 0) => {
             for item in &mut avg_sparse.values {
                 *item /= sparse_count as VectorElementType;
             }
-            Ok(avg_sparse.into())
+            Ok(Vector::from(avg_sparse))
         }
-        (_, _) => Err(CollectionError::bad_input(
-            "Can't average dense and sparse vectors together".to_owned(),
+        (0, 0, _) => Ok(Vector::from(avg_multi.unwrap())),
```

Comment:
maybe we could do it without the `unwrap` :)

Reply:
There are no default multivector to replace unwrap. And placeholder cannot be constructed because if `avg_multi` is empty, dimension is unknown here. I can only return `service_error` instead

Reply:
fixed